src/frontends/paddle/src/op/p_norm.cpp

```diff
@@ -0,0 +1,43 @@
+// Copyright (C) 2018-2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "default_opset.hpp"
+#include "openvino/frontend/paddle/node_context.hpp"
+
+namespace ov {
+namespace frontend {
+namespace paddle {
+namespace op {
+NamedOutputs p_norm(const NodeContext& node) {
+    auto data = node.get_input("X");
+    auto p = node.get_attribute<float>("porder", 2.0);
+    auto axis = node.get_attribute<int32_t>("axis", -1);
+    auto keepdim = node.get_attribute<bool>("keepdim", false);
+
+    auto absNode = std::make_shared<default_opset::Abs>(data);
+    auto axisNode = default_opset::Constant::create(ngraph::element::i32, {1}, {axis});
+
+    if (p == std::numeric_limits<float>::infinity()) {
+        return node.default_single_output_mapping(
+            {std::make_shared<default_opset::ReduceMax>(absNode, axisNode, keepdim)},
+            {"Out"});
+    } else if (p == -std::numeric_limits<float>::infinity()) {
+        return node.default_single_output_mapping(
+            {std::make_shared<default_opset::ReduceMin>(absNode, axisNode, keepdim)},
+            {"Out"});
+    } else {
```

Comment:
looks like we also need to separately process "(p == 0)" case here, because 'Power' Op can't handle '(p = 0) && input tensor(data) contains 0 values' cases. Maybe we can use 'NonZero' Op with 'get_partial_shape()' function(please use PADDLE_OP_CHECK to check that its rank is static before using the following 'Gather' Op) and 'Gather' Op for "(p == 0)" case.

Reply:
Hi,
The second test case contains the situation that p=0:
```python
    input_shape = (3, 4, 5)
    input_data = np.random.rand(*input_shape).astype(np.float32)
    paddle_result = p_norm('p_norm2', input_data, axis=0, p=0.0, keepdim=None)
```
And I used all-zero tensor to test locally, the fuzzy test can still pass:
![image](https://user-images.githubusercontent.com/72954905/183294611-6fcf6c50-942b-4b57-9a2e-b59b60bd3189.png)
I am not sure that whether need to separately process p=0 or input tensor that contains zeros

Reply:
I mean in the the single test case that contains both 'input_data' contains 0 values and p=0. (e.g. input_data=np.zeros(input_shape).astype(np.float32) 
paddle_result=p_norm('p_norm6'), input_data, axis=0, p=0, keepdim=None) )
could you please add such an additional test case and have a try?

Reply:
Hi,
You are right, so I add the  extra branch to process the situation that p==0 and input data contains 0 values simultaneously.
I use the `NotEqual` op to get a bool matrix;
Then convert it to the input data type;
Finally, use `ReduceSum` op to get result; 
The tests passed:
![image](https://user-images.githubusercontent.com/72954905/183466842-91675d45-b323-46b7-9f42-12aba2a51948.png)

I also follow the steps you mentioned above for clang-format, but i am not sure whether it will work. I have

Reply:
Hi, @Asthestarsfalll : From the algorithm level, it's quite good to use 'NotEqual' and 'ReduceSum' way to process the situation that p==0 and input data contains 0 values. But I think we still need to handle input data rank is 1 situation in this process way. e.g. 
    _input_data = np.array([[0, 1, 2, -10]]).astype("float32")
    paddle_result = p_norm('p_norm7', input_data, axis=0, p=0.0, keepdim=False)_

In this kind of situation, openvino 'ReduceSum' output shape will be null, while Paddle final output shape is {1}. Maybe we can add an 'Reshape'(to shape {1}) OP after 'ReduceSum' for such kind of situation.
By the way, could you please also add a such above kind of test case?

For clang-format issue, you may try the following two ways to fix it:
(1) if you use VS code IDE, you may go to the failed 'paddle/src/op/p_norm.cpp' file, 'Select all' -> Right click to select 'Format Selection'.
(2) or go to openvino build folder, run 'make clang_format_fix_all' command(before that, make sure that you have installed 'clang-format-9' and used '-DENABLE_CLANG_FORMAT=ON' cmake option during build openvino).

Reply:
Down